Expression evaluator

VinInn opened the pull request:
A new Expression Evaluator based on the standard C++ compiler
to replace the current one that is implemented using on a home-made parser (based on boost::spirit) and access objects using ROOT-reflection

motivations, doc, examples in 
http://innocent.home.cern.ch/innocent/ExprEval.pdf
to be presented at next reco/phys-tool meeting

davidlange6 committed:
round one of migration

Forward-ported from commit 3eb2eb0a of #1026

davidlange6 committed:
migration test/NavigationSchoolAnalyzer.cc

Forward-ported from commit eeed3e17 of #1026

davidlange6 committed:
detid migration

Forward-ported from commit 6ccd2da0 of #1265.

davidlange6 committed:
tracker detid migration

Forward-ported from commit 46379045 of #1265

fruboes committed:
New double jet filter - allow specific dijet topologies in eta

fruboes committed:
Single jet filter with eta min/max capability

fruboes committed:
More flexibility in pt requirements

Dr15Jones committed:
Do not call edm::RefVector<>::product in Associators

The call to edm::RefVector<>::product is being deprecated. Instead
of getting the full container, we now loop over the RefVector and
fill a new container. This (argueably) better matches the API of
the function call anyway.
This may lead to differences is results because we might be looping
over a restricted set of entries. If this is a problem then the calling
module should pass in the full container and not a RefVector.

yetkinyilmaz committed:
gen-sequences and modules standardized

yetkinyilmaz committed:
gen-sequences and modules standardized

yetkinyilmaz committed:
subevent ON but mixing OFF

yetkinyilmaz committed:
hiGenJets also use genParticlesForJets

fruboes committed:
Fix jet counting

fruboes committed:
Fix defaulModuleLabel

fruboes committed:
Fix constness, fix missing initialization

fruboes committed:
More constness cleanup

yetkinyilmaz committed:
more clean-up on old obsolete himix stuff

yetkinyilmaz committed:
more clean-up on old obsolete himix stuff

yetkinyilmaz committed:
mix vs non-mix swapped

yetkinyilmaz committed:
pp-only workflows added back

wmtan committed:
Add mssing link dependency and fix checksum

davidlange6 committed:
Merge pull request #7248 from davidlt/clang-fix-Calibration-IsolatedParticles-v2

Calibration/IsolatedParticles: definite static const int members passed by a ref

davidlange6 committed:
Merge pull request #7250 from davidlt/clang-fix-CalibFormats-CaloObjects-v2

CalibFormats/CaloObjects: definite static const int members passed by a ref

davidlange6 committed:
Merge pull request #7280 from Martin-Grunewald/FixOldL1RePack

Fix old L1RePack file to cure 'unRunnable Schedule'

cmsbuild committed:
Merge pull request #7293 from cms-tsg-storm/goingOnWithHltUpdatesForRun2

Further HLT updates for Run2

cmsbuild committed:
Merge pull request #7282 from ndaci/ExoValDev_74Xto73X_PR2

Fix commas. Backport from 74X to 73X.

cmsbuild committed:
Merge pull request #7263 from silviodonato/HLTJetTagWithMatching_73X

Backport of "Fixing typo in HLTJetTagWithMatching #7261 "

wmtan committed:
Remove circular dependency

wmtan committed:
Fix cyclic dependency and checksum

lihux25 committed:
Adjust code to allow one pulse fit for high ampl > 500 fC.

cmsbuild committed:
Merge pull request #7281 from fruboes/NewFSQJetFiltes_74x_20150120_0200

New fsq jet filtes 74x 20150120 0200

makortel committed:
Move G4Event to TLS

G4Event holds thread-local stuff, of which one is G4Allocator. If
G4Event is a class member, it may happen that the G4Event (and thus
G4Allocator) get destructed in a different thread it was constructed.
Their destructors may even be executed twice in a thread resulting a
segfault. Moving G4Event to TLS fixes the issue.

Commit:
loop bugfix

fruboes committed:
Revert "Fix defaulModuleLabel"

This reverts commit ab38c63ed8dd63c2a321052b50b8d223b18fcfce.

fruboes committed:
Finish const fixes

fruboes committed:
Finish const fixes

Commit:
Correcting patch to use fragment from Mikhail Kirsanov upon his request

Commit:
filter cfi prepared

Commit:
test cfg files added

cmsbuild committed:
Merge pull request #7238 from davidlt/clang-fix-CondTools-Ecal-v2

CondTools/Ecal: drop VLA of non-POD type

Commit:
a few fixes, and some synchronzation of cfg structure with fullsim

Commit:
adding Mikhail's patch to the TauolaInterface

Commit:
old comments implemented

Commit:
Keep Entry::mean in DB object

Shape change is not allowed

Commit:
Add entry.mean again

Commit:
Forgot ;

Commit:
merged

cmsbuild committed:
Merge pull request #7300 from fruboes/NewFSQJetFiltes_74x_20150120_0200

Finish const fixes

aledegano committed:
Merge pull request #7290 from davidlange6/new_hf_SL

bump to v3 of both hf SL

yetkinyilmaz committed:
HI 2.76 TeV nominal vtx smearing added back

yetkinyilmaz committed:
HI 2.76 TeV nominal vtx smearing added back

vanbesien committed:
ClusterCount now using DQMEDAnalyzer instead of EDAnalyzer

cmsbuild committed:
Merge pull request #7288 from franzoni/NominalCollision2015-default-newGS-namingChanges-CMSSW_7_4_0_pre5

NominalCollision2015 beamspot: migration of SIM and RECO BS

yetkinyilmaz committed:
save the same playbackinfo as latest when running playback

yetkinyilmaz committed:
save the same playbackinfo as latest when running playback

yetkinyilmaz committed:
hiGenParticles obsolete, using genParticles

yetkinyilmaz committed:
hiGenParticles obsolete, using genParticles

BetterWang committed:
Pythia8 update.

cmsbuild committed:
Merge pull request #7209 from anorkus/threaded_dqm_migration4

Threaded dqm migration4

Commit:
more

Commit:
centrality update to event plane code

cmsbuild committed:
Merge pull request #7303 from makortel/geant4mt_fix_G4Event

Move G4Event to TLS in RunManagerMTWorker

fwyzard committed:
FastTimerService: thread-safe fix in preStreamBeginRun

boudoul committed:
particle guns (pythia8) for upgrade

cmsbuild committed:
Merge pull request #7291 from wmtan/AddMissingLinkDepenedency

Add mssing link dependency and fix checksum

cmsbuild committed:
Merge pull request #7298 from wmtan/RemoveCircularDependencyAndFixChecksum

Remove circular dependency and fix checksum for class reco::CATopJetTagInfo.

boudoul committed:
aligned with SLHC

boudoul committed:
aligned with SLHC

boudoul committed:
aligned with SLHC

boudoul committed:
aligned with SLHC

boudoul committed:
aligned with SLHC

boudoul committed:
aligned with SLHC

boudoul committed:
Preparing Upgrade WFs in RunTheMatrix

ahinzmann committed:
add ak5 castor jets

cmsbuild committed:
Merge pull request #7289 from dmitrijus/dqm_multithread_migration_a4

Migration to the thread-safe DQM: #4

vanbesien committed:
Added empty implementation of bookHistograms

Commit:
- disable track mixing in case of gen-level mixing
- correct label for pileup information in case of gen-level mixing

ahinzmann committed:
add ak5 castor jets part 2

archiron committed:
false lines removed : setBookPrefix and setBookIndex

mmachet committed:
changed the code structure for prompt photons definition

Commit:
Make doxygen comments refer to the correct thing

Commit:
Pass EncodedEventId to TrackingVertex constructor instead of EncodedTruthId

VinInn committed:
added new ExpressionEvaluator and its unit test

VinInn committed:
added new ExpressionEvaluator and its unit test

VinInn committed:
added new ExpressionEvaluator and its unit test

yetkinyilmaz committed:
centrality input label for flat evtplane

VinInn committed:
added test infrastructure for real Candidate

Commit:
complete the event content for fastsim pu files

arizzi committed:
adding py config for PF version, example in test/ and finishing association stuff

arizzi committed:
remove cout

bendavid committed:
streamline herwigpp common fragments and add examples for lhe and pure herwig fragments

VinInn committed:
test on reco::Candidate successful

perrotta committed:
Updathe the HLT menu up to the /dev/CMSSW_7_3_0/HLT/V80

perrotta committed:
Updathe the HLT menu up to the /dev/CMSSW_7_3_0/HLT/V80

cmsbuild committed:
Merge pull request #7314 from lihux25/method2_extend_one_pulse_fit_to_high_ampl

Adjust Method 2 code to allow one pulse fit for high ampl > 500 fC

VinInn committed:
clean include area as well

deguio committed:
add dqmio option to promptReco and harvesting scripts

deguio committed:
add dqmio option to promptReco and harvesting scripts

ahinzmann committed:
get rid of tau and HLT changes

cmsbuild committed:
Merge remote branch 'origin/CMSSW_7_3_X' into CMSSW_7_0_X

gartung committed:
Update script the generates list of data classes and update bloom filter
resulting from the new class list. The script starts with the list of data
classes and all derived classes. For each class the member data classes and
derived classes are added to the list. 10 iterations are perfomed on the
list so that as many levels of indirection are covered.

Dr15Jones committed:
Make ProcForeach used by MVAComputer thread safe

ProcForeach was using mutable member data to keep track of the
state of the looping it was doing. Given that multiple threads
could be updating those values concurrently, this lead to crashes.
The code has now been changed to store the looping state externally
via VarProcessor::LoopCtx on the stack inside MVAComputer::evalInternal.
This allows each thread to have its own copy.

Dr15Jones committed:
Minor cleanup of AtomicId

Theoretically, making the allocator static could lead to an order
of initialization problem. In reality the default allocator is
stateless so wasn't a problem. However not being static simplifies
understanding the code.
Since non of the member functions are const we don't need to use
mutable.

Dr15Jones committed:
Make ProcessRegistry usage thread-safe

The static analyzer found that the registries were being accessed
at event processing time by some modules. Given that we could get
a shared library load happening while some other code was reading
the registry, we needed to make the lookup and registration process
threads safe. This was easy to accomplish using tbb::concurrent_unordered_map.

gartung committed:
Change dyn_cast to dyn_cast_or_null to avoid EOF error during cast
This kept PhysTools::ProcForeach from being analyzed.

davidlange6 committed:
Merge pull request #7175 from Dr15Jones/eventBasedTrackAssociation

Event based track and muon to simulation associators

gartung committed:
null in this case would be bad

gartung committed:
Merge branch 'CMSSW_7_4_X' of github.com:gartung/cmssw into clangSA-update-classname-bloom-filter

cmsbuild committed:
Merge pull request #7307 from fwyzard/FastTimerService_fix_preStreamBeginRun

FastTimerService: thread-safe fix in preStreamBeginRun

cmsbuild committed:
Merge pull request #7313 from boudoul/UPWFS

Preparing Upgrade WFs in RunTheMatrix

cmsbuild committed:
Merge pull request #7309 from boudoul/genforupgradev2

particle guns (pythia8) for upgrade

cmsbuild committed:
Merge pull request #7304 from vanbesien/migrating_ClusterCount_in_SiStripCommissioningSources

Migrating ClusterCount in SiStripCommissioningSources to new DQM API

cmsbuild committed:
Merge pull request #7316 from mark-grimes/modifyTrackingParticleDocs

Minor documentation change to TrackingParticle

cmsbuild committed:
Merge pull request #7324 from bendavid/herwigcfg_74x

Further Herwig++ config cleanup

cmsbuild committed:
Merge pull request #7326 from cms-tsg-storm/furtherUpdatesToTheRun2HltIntegration

Further updates to the Run2 HLT menu

cmsbuild committed:
Merge pull request #7330 from gartung/clangSA-update-classname-bloom-filter

Update script the generates list of data classes and update bloom filter

cmsbuild committed:
Merge pull request #7272 from nickmccoll/MoveToTrackerTopology

Removal of references to depreciated SiStripDetId

cmsbuild committed:
Merge pull request #7318 from mark-grimes/fixTrackingTruthAccumlator

Fix some obscure code in TrackingTruthAccumulator (still functionally identical)

Martin-Grunewald committed:
Cleanup of HLT configs

cmsbuild committed:
Merge pull request #7057 from UAEDF-tomc/weightedLikelihood

Clean up likelihood algorithm (parts of code now executed before creation of database object) + updated DBWriter

covarell committed:
remove empty source and fix Hgg decay

yetkinyilmaz committed:
Centrality DQM fix for new centrality

Commit:
New Global Tags for Release Validation in all the supported scenarios.

covarell committed:
add Z mass limits

VinInn committed:
add analyzer example

cmsbuild committed:
Merge pull request #7336 from Martin-Grunewald/HLTcleanup

Cleanup of HLT configs

cmsbuild committed:
Merge pull request #7302 from lveldere/fastsim-new-default-mix-scheme

Fastsim new default mix scheme

davidlange6 committed:
Merge pull request #7333 from Dr15Jones/doNotCallRefVector_productInAssociators

Do not call edm::RefVector<>::product in Associators

Dr15Jones committed:
Remove unused variables from test module

davidlange6 committed:
Merge pull request #7327 from deguio/T0scripts

add dqmio option to promptReco and harvesting scripts

ktf committed:
Merge pull request #7340 from Dr15Jones/removeUnusedVariablesFromTest

Remove unused variables from test module

VinInn committed:
test also selector, factorize

yetkinyilmaz committed:
UETable, without the corresponding reco, added

Commit:
bug fix phi meson

yetkinyilmaz committed:
more members in UETable

cmsbuild committed:
Merge remote branch 'origin/CMSSW_7_3_X' into CMSSW_7_0_X

wmtan committed:
Remove obsolete unused typedefs

gartung committed:
Merge branch 'CMSSW_7_4_X' of https://github.com/cms-sw/cmssw into clangSA-update-classname-bloom-filter

Martin-Grunewald committed:
L1Skim Fix

cmsbuild committed:
Merge pull request #7339 from diguida/autoCond_740pre6

New Global Tags for Release Validation in all the supported scenarios.

bendavid committed:
change weight accessor to conform to HepMC standard

cmsbuild committed:
Merge pull request #7344 from Martin-Grunewald/L1SkimFix

L1Skim Fix for HLT

bendavid committed:
fix weights for multiple hadronization to conform to hepmc specification

bendavid committed:
fix weights for pythia8 ckkw merging to conform to hepmc specification

bendavid committed:
fix merging weight for Pythia8 UMEPS/UNLOPS

Commit:
DQM update for 74X for 7e33 triggers

Commit:
DQM update for 74X for 7e33 triggers

Commit:
DQM update for 74X for 7e33 triggers

bendavid committed:
fill lhe weights manually for herwig++

cmsbuild committed:
Merge pull request #7193 from mtosi/from-CMSSW_7_4_0_pre3_addESmoduleTEMPLATE

add ESModule template

gartung committed:
Use python script instead of awk script to find member classes and derived classes of event setup data classes

cmsbuild committed:
Merge pull request #7299 from fruboes/NewFSQJetFiltes_73x_20150121_0200

New fsq jet filtes - backport to 73

cmsbuild committed:
Merge pull request #7348 from bendavid/hepmcweightfix

Fix handling of weights in generator interfaces

gartung committed:
Update run_class_dumper.sh

gartung committed:
+x permissikons on class-composition.py

cmsbuild committed:
Merge pull request #7350 from bendavid/herwigppweightfix

fill lhe weights manually for herwig++

cmsbuild committed:
Merge pull request #7351 from gartung/clangSA-update-classname-bloom-filter

Clang sa update ESD class list generation

Dr15Jones committed:
Added unit test for MVAComputer

yetkinyilmaz committed:
added override to beginRun()

Dr15Jones committed:
bug fix: ProcForeach check condition needed a 'not'

When initializing the value of 'size' for the LoopCtx we need to
check that the 'size' has not yet been set. The bug only changed
the value of 'size' if it had already been set.

gartung committed:
use functions-calls-db.txt instead of db.txt

Commit:
workflow ZMM_13TeV_TuneCUETP8M1_cfi also in fast sim

Commit:
same ZTT GEN for fast and full sim

Commit:
re-number ZMMFS_13 to be 135.9; no perf change

cmsbuild committed:
Merge pull request #7354 from franzoni/fast-sim-zmm-ztt-CMSSW_7_4_X_2015-01-23-0200

Fast sim zmm ztt

cmsbuild committed:
Merge pull request #7353 from gartung/clangSA-update-classname-bloom-filter

use functions-calls-db.txt instead of db.txt

bendavid committed:
Merged 7_4_X_EvtGenUndecayPatch from repository inugent

bendavid committed:
switch cout to LogWarning

VinInn committed:
use standard templates evaluators

VinInn committed:
add dir listing

mmachet committed:
fixed compilation error

ktf committed:
Merge pull request #7332 from Dr15Jones/fixThreadSafetyOfMVAComputer

Fix thread safety problem in PhysicsTools/MVAComputer

htrauger committed:
Revised code to make pixel residual plots by layer

cmsbuild committed:
Merge pull request #7338 from covarell/74X_removeEmpty

remove empty source and fix Hgg decay

cmsbuild committed:
Merge pull request #7355 from bendavid/evtgentauoladecayfix

Evtgen / Tauola++ decay fixes

cmsbuild committed:
Merge pull request #6976 from mmachet/CMSSW_7_4_X

made EM enriched filter more indipendent from the generator used and add...

cmsbuild committed:
Merge pull request #7342 from wmtan/RemoveUnusedObsoleteTypedefs

Remove obsolete unused typedefs

Commit:
fastsim: digi-reco mixing: fixes after detailed review of effects on dumped fastsim cfgs

Commit:
fastsim: fix unit tests after transition to digi-reco mixing as default (unit tests stay on gen-mixing for the moment)

Dr15Jones committed:
Make the static EgCutCodes::codes_ const

Dr15Jones committed:
EgHLTOfflineSource is not thread safe

The module uses the non-const static egHLT::TrigCodes::trigBitSetMap_.

wmtford committed:
Fix bug chosing closest simHit

ahinzmann committed:
remove commented lines

cmsbuild committed:
Merge pull request #7312 from boudoul/updating2017

porting 2017 scenario in 74X : first round of changes

ahinzmann committed:
remaining ak5 to ak4 transitions in DQM and HLT

davidlange6 committed:
Merge pull request #7335 from makortel/phase1_fwd_TrackerTopology

Forward-port TrackerTopology migration from 620_SLHC

VinInn committed:
fix unit test, implement and test altermate syntax including macro

cmsbuild commented:
A new Pull Request was created by @VinInn (Vincenzo Innocente) for CMSSW_7_4_X.

Expression evaluator

It involves the following packages:

CommonTools/Utils


@nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.

degano commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Commit:
update the list of paths

cmsbuild committed:
Merge pull request #7362 from wmtford/pixRecHitValid

Fix bug chosing closest simHit

cmsbuild commented:
-1 
 
Tested at: c425846d5ff9672746c8d8234b00f044ce079a57
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test ExpressionEvaluatorUnitTest had ERRORS
---> test testExpressionEvaluator had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2117/summary.html

VinInn committed:
fix to run non under src

VinInn commented:
@cmsbuild ,
ok let's try again...

cmsbuild commented:
Pull request #7363 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cmsbuild committed:
Merge pull request #7341 from lucamartini/Quarkonia7E33DQM

bug fix phi meson in heavy flavor HLT offline DQM

degano commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: e22bf0b4d521e87f3b97a32d66a685cd95b073ec
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test testExpressionEvaluator had ERRORS
---> test ExpressionEvaluatorUnitTest had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2118/summary.html

Commit:
- recovered option --pileup default for cmsDriver
- default input files for gen-mixing are now 8 TeV
- when postLS1 customisation is used, input files for gen-mixing are 13 TeV
- HLT step runs independent from RECO step (broken during transition to digi-reco mixing)

cmsbuild committed:
Merge pull request #7346 from CmsHI/centrality_74X_03

Centrality proper migration to consumes

davidlange6 committed:
Merge pull request #7329 from ahinzmann/RemoveJetCollections

Update jet AOD/RECO event content

cmsbuild committed:
Merge pull request #7349 from jmduarte/razor-run2-hlt-7e33

Razor run2 hlt 7e33

VinInn committed:
clean area before start

VinInn commented:
to avoid polluting the release with test packages I am trying to create and compile it on the fly with

```
   system("rm -rf $CMSSW_BASE/src/VITest $CMSSW_BASE/include/$SCRAM_ARCH/VITest");
   system("echo $CMSSW_BASE");
   system("mkdir -p $CMSSW_BASE/src/VITest/ExprEval/src; cp $CMSSW_BASE/src/CommonTools/Utils/test/ExprEvalStubs/*.h $CMSSW_BASE/src/VITest/ExprEval/src/.");
   system("cp $CMSSW_BASE/src/CommonTools/Utils/test/ExprEvalStubs/BuildFile.xml $CMSSW_BASE/src/VITest/ExprEval/.; pushd $CMSSW_BASE; scram b -j 8; popd");
   system("ls $CMSSW_BASE/src; ls -l $CMSSW_BASE/src/VITest/ExprEval/src");
   system("ls -l $CMSSW_BASE/include/$SCRAM_ARCH/VITest/ExprEval/src");
```

need to understand what goes wrong in jenkins build w/r/t/ my own area

cmsbuild commented:
Pull request #7363 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

Dr15Jones committed:
Make pointers in static arrays const

The static analyzer complained that the pointers in the const arrays
were not const. Changing them to const did not change the semantics
of the algorithm.

ktf committed:
Merge pull request #7369 from Dr15Jones/makePointersInArraysConst

Make pointers in static arrays const

perrotta committed:
Another bunch of updates for the Run2 HLT menu

Dr15Jones committed:
Change static 'debug' to constexpr

The static analyzer was complaining about the non-const static 'debug'.
Given the value is set once and never changes, it was changed to be a
constexpr.

VinInn commented:
@cmsbuild please test

ianna committed:
Fix RPC validation tests

ktf committed:
Merge pull request #7371 from Dr15Jones/makeDebugStaticConstexpr

Change static 'debug' to constexpr

mdhildreth committed:
fixes to CSC Digitization/Trigger paths for PreMixing

davidlange6 committed:
Merge pull request #7306 from lveldere/fastsim-cmsdriver-digirecomix

Fastsim cmsdriver digirecomix

davidlange6 committed:
Merge pull request #7373 from mdhildreth/74X_PreMix_FixCSC

74X fixes to CSC Digitization/Trigger paths for PreMixing

davidlange6 committed:
Merge pull request #7358 from lveldere/fastsim-digirecomix-fixes

Fastsim digirecomix fixes

perrotta committed:
Fix JetTag in validation configs

davidlange6 committed:
Merge pull request #7138 from BetterWang/ZEEMM_13_HI_74X

ZEEMM_13

cmsbuild committed:
Merge pull request #7328 from deguio/T0scripts_74

add dqmio option to promptReco and harvesting scripts

perrotta committed:
Fix JetTag in validation configs

perrotta committed:
Adapt the RelVal EventContent to the renamed BTag modules

cmsbuild committed:
Merge pull request #7372 from ianna/rpc-validation-tests

Fix RPC Validation Tests

Dr15Jones committed:
Use thread safe container for EventSetup type lookup

The static analyzer discovered that the type lookup function was being called at event loop time. If a shared library was loaded during the event loop it could lead to a data race. Switching to a tbb thread safe container avoids the race condition.

yetkinyilmaz committed:
merged recent updates to 74X

davidlange6 committed:
Merge pull request #7285 from arizzi/pvSorting-from740p3-enabled

Introduce better PV sorting

cmsbuild committed:
Merge pull request #7375 from Dr15Jones/useThreadSafeContainerForTypelookup

Use thread safe container for EventSetup type lookup

cmsbuild committed:
Merge pull request #7357 from htrauger/from-CMSSW_7_4_X_2015-01-24-0200

Revised code to make pixel residual plots by layer

Dr15Jones committed:
removed unnecessary mutable designations

The static analyzer complained about thread-unsafe mutables in the
class. As the mutable was unnecessary, it was removed.

perrotta committed:
Fix module name also in the Fake menu

ktf committed:
Merge pull request #7379 from Dr15Jones/removeUnnecessaryMutablesFromHcalDcsValues

removed unnecessary mutable designations from HcalDcsValues

davidlange6 committed:
Merged refs/pull/7287/head from repository cms-sw

davidlange6 committed:
Merge pull request #7287 from CmsHI/mix_74X_11

Bugfixes and clean-up after HI mixing updates

davidlange6 committed:
add separate pretrking sequences for vertexreco to use

davidlange6 committed:
Merge pull request #7381 from davidlange6/fastsim_pretrkcalo

Fastsim pretrkcalo

slehti committed:
Updated h2tau electron paths to WP75

cmsbuild committed:
Merge pull request #7370 from cms-tsg-storm/hltUpdatesAsOf150126

Another bunch of updates for the Run2 HLT menu

cmsbuild committed:
Merge pull request #7360 from Dr15Jones/fixedThreadProblemsInDQMOfflineTrigger

Fixed thread problems in DQMOffline/Trigger

perrotta committed:
Fix with the correct names of the PF-based alpha-T trigger paths as currently in the Run2 menu

degano commented:
@cmsbuild please test

VinInn commented:
Wait, discovered a concurrency issue when running test in parallel....
fixing code right now...

cmsbuild commented:
The tests are being triggered in jenkins.

ianna committed:
Use Extended as a default geometry for validation

ianna committed:
Use Extended as a default geometry for validation

VinInn committed:
avoid clash between tests run in parallel

cmsbuild commented:
Pull request #7363 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

VinInn commented:
@cmsbuild please test

this time tests running in parallel should not step on each other toes

degano commented:
@cmsbuild please test
Aborted the previous tests, restarting now.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: bb415e2f5e614af11d6a7da129beefed27b98e5e
 When I ran the RelVals I found an error in the following worklfows: 
 5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

----- Begin Fatal Exception 27-Jan-2015 13:35:42 CET-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 1 lumi: 1 event: 1

   [1] Running path 'simulation_step'

   [2] Calling event method for module HBHEIsolatedNoiseReflagger/'hbhereco'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<reco::TrackExtrapolation>

Looking for module label: trackExtrapolator

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

----- Begin Fatal Exception 27-Jan-2015 13:36:36 CET-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 1 lumi: 1 event: 1

   [1] Running path 'simulation_step'

   [2] Calling event method for module HBHEIsolatedNoiseReflagger/'hbhereco'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<reco::TrackExtrapolation>

Looking for module label: trackExtrapolator

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2144/summary.html

VinInn commented:
cannot be...
@cmsbuild please test
run unit tests
relvals CANNOT be affected by this PR

slava77 commented:
We need a bit better error detection/interpretation in the jenkins/cms-bot tests.
If the same workflow is broken in the reference IB, the error should at least have a note of that.
Otherwise it triggers too much noise that takes an effort to interpret for people who don't watch the whole cmssw repo.
@ktf @nclopezo @degano 
Please check or follow up.
Thank you.

degano commented:
@slava77 @VinInn this is indeed a problem with the latest IB (2015-01-27-0200). 
I'll re-trigger the tests once the afternoon IB is available (where that issue has been fixed).

davidlange6 commented:
I agree - maybe some simple mask can be added that compares against the baseline (ok, this implies having a baseline though)

On Jan 27, 2015, at 3:32 PM, Slava Krutelyov <notifications@github.com>
 wrote:

> We need a bit better error detection/interpretation in the jenkins/cms-bot tests.
> If the same workflow is broken in the reference IB, the error should at least have a note of that.
> Otherwise it triggers too much noise that takes an effort to interpret for people who don't watch the whole cmssw repo.
> @ktf @nclopezo @degano 
> Please check or follow up.
> Thank you.
> 
> —
> Reply to this email directly or view it on GitHub.
>

cmsbuild committed:
Merge pull request #7347 from CmsHI/uetable_74X_01

New conditions objects for HI jet reconstruction

cmsbuild committed:
Merge pull request #7296 from olivito/smpdqm_photon_73x

update photon path names in SMP HLT DQM config

davidlange6 committed:
Merge pull request #7386 from ianna/default-geometry-validation

Default Geometry Validation

davidlange6 committed:
Merge pull request #7364 from HuguesBrun/updateHiggsDQMin740pre5

update the list of paths monitored by Higgs HLT DQM

davidlange6 committed:
Merge pull request #7315 from archiron/ElectronAnalyzer_Mistake

ElectronAnalyzer.cc  : false lines removed - setBookPrefix and setBookIndex

davidlange6 committed:
Merge pull request #7383 from slehti/HiggsValidationUpdate_CMSSW74X_27012015

Updated h2tau electron paths to WP75

davidlange6 committed:
Merge pull request #7385 from perrotta/fixPathNamesDiPFJetIntoPFDiJet

Fix with the correct names of the PF-based alpha-T trigger paths

cmsbuild committed:
Merge remote branch 'origin/CMSSW_7_3_X' into CMSSW_7_0_X

cmsbuild committed:
Merge pull request #7407 from fwyzard/edmPythonConfigToCppValidation_update_functionality

update edmPythonConfigToCppValidation functionality

davidlange6 commented:
please test

degano commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: bb415e2f5e614af11d6a7da129beefed27b98e5e
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test ExpressionEvaluatorUnitTest had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2157/summary.html

VinInn commented:
doing scram b from inside each given test does not work, they clash
need to find a different approach (at least this time one of the two succeeded...)

smuzaffar commented:
@VinInn , yes doing "scram build" within a "scram build" is not a right approach. I think right approach should be to move EEUnitTest/ExprEval/src/*.h files in to CommonTools/Utils/src and let scram compile then when you do first scram build.

VinInn commented:
I was hoping NOT to pollute the release in particular with a huge gch used only for my tests.
In addition it may create circular dependencies...
what I will do is to create a subsystem ExprEvalTest with all the files already in there
we will then decide how to make it "appearing" during the tests if we do not want to have it around in the distribution...

VinInn committed:
add test directory as separate package

VinInn committed:
add test directory as separate package

VinInn committed:
tests run

VinInn commented:
@cmsbuild please test

cmsbuild commented:
Pull request #7363 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

VinInn commented:
Now the test package is in the "release"
I am sure we can find a way to hide it somewhere and materialize it only at scram b runtests time...

smuzaffar commented:
let me check if we can do scram b with in scram b runtests

smuzaffar commented:
I think if you do something like the following then it should work
- change CommonTools/Utils/test/BuildFile.xml to have
<bin file="ExpressionEvaluatorUnitTest.cpp">
  <flag TEST_RUNNER_CMD="src/CommonTools/Utils/test/ExpressionEvaluatorUnitTest.sh"/>
  <use name="CommonTools/Utils"/>
</bin>

- create CommonTools/Utils/test/ExpressionEvaluatorUnitTest.sh script (with execution permission) which should copy the $CMSSW_BASE/src/CommonTools/Utils/test/ExprEvalStubs contents in to  $CMSSW_BASE/src/EEUnitTest/ExprEval/src and then do "scram build EEUnitTest/ExprEval"

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2157/summary.html

ktf commented:
@cmsbuild please test

degano commented:
Test comment

VinInn commented:
the real problem is that I have two tests:
```
<bin   name="testExpressionEvaluator" file="testExpressionEvaluator.cc,testRunner.cpp">
  <flags TEST_RUNNER_CMD="src/CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh"/>

<bin file="ExpressionEvaluatorUnitTest.cpp">
<flags TEST_RUNNER_CMD="src/CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh"/>

```

with
```
cat CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh
echo running ExpressionEvaluatorIntTest
mkdir -p $CMSSW_BASE/src/ExpressionEvaluatorTests
cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/EEIntTest $CMSSW_BASE/src/ExpressionEvaluatorTests/EEIntTest
scram b ExpressionEvaluatorTests/EEIntTest
ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/EEIntTest/src


[innocent@vinavx2 src]$ cat CommonTools/Utils/test/ExpressionEvaluatorUnitTest.sh 
echo running ExpressionEvaluatorUnitTest
mkdir -p $CMSSW_BASE/src/ExpressionEvaluatorTests
cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/EEUnitTest $CMSSW_BASE/src/ExpressionEvaluatorTests/EEUnitTest
scram b ExpressionEvaluatorTests/EEUnitTest
ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/EEUnitTest/src

```

the two "scram b" interferes 

is it possible to have just one script that runs ONCE?
what I need to do is
```
cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/ $CMSSW_BASE/src/ExpressionEvaluatorTests
scram b ExpressionEvaluatorTests
ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/*/src
$CMSSW_BASE/test/$SCRAM_ARCH/ExpressionEvaluatorUnitTest
$CMSSW_BASE/test/$SCRAM_ARCH/testExpressionEvaluator
rm -rf  $CMSSW_BASE/src/ExpressionEvaluatorTests
```

cmsbuild commented:
The tests are being triggered in jenkins.

smuzaffar commented:
I remember now that, while running runtests/unittests during IB 
validation, we disallow "scram b" otherwise it will rebuild every thing 
(remember we install release using RPM at the validation time).

On 1/28/2015 11:54 AM, Vincenzo Innocente wrote:
>
> Now the test package is in the "release"
> I am sure we can find a way to hide it somewhere and materialize it 
> only at scram b runtests time...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/7363#issuecomment-71815213>.
>

smuzaffar commented:
yes, it is possible to run only one tests. You need to disable one test 
by adding the NO_TESTRUN flag e.g.|
<bin name="testExpressionEvaluator" 
file="testExpressionEvaluator.cc,testRunner.cpp">|
   <flags NO_TESTRUN="1"/>
</bin>

but as I mentioned these tests will only run if they are part of release 
itself (scram build during IB tests will not compile any new test).

On 1/28/2015 2:24 PM, Vincenzo Innocente wrote:
>
> the real problem is that I have two tests:
>
> |<bin   name="testExpressionEvaluator" file="testExpressionEvaluator.cc,testRunner.cpp">
>    <flags TEST_RUNNER_CMD="src/CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh"/>
>
> <bin file="ExpressionEvaluatorUnitTest.cpp">
> <flags TEST_RUNNER_CMD="src/CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh"/>
>
> |
>
> with
>
> |cat CommonTools/Utils/test/ExpressionEvaluatorIntTest.sh
> echo running ExpressionEvaluatorIntTest
> mkdir -p $CMSSW_BASE/src/ExpressionEvaluatorTests
> cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/EEIntTest $CMSSW_BASE/src/ExpressionEvaluatorTests/EEIntTest
> scram b ExpressionEvaluatorTests/EEIntTest
> ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/EEIntTest/src
>
>
> [innocent@vinavx2 src]$ cat CommonTools/Utils/test/ExpressionEvaluatorUnitTest.sh
> echo running ExpressionEvaluatorUnitTest
> mkdir -p $CMSSW_BASE/src/ExpressionEvaluatorTests
> cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/EEUnitTest $CMSSW_BASE/src/ExpressionEvaluatorTests/EEUnitTest
> scram b ExpressionEvaluatorTests/EEUnitTest
> ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/EEUnitTest/src
>
> |
>
> the two "scram b" interferes
>
> is it possible to have just one script that runs ONCE?
> what I need to do is
>
> |cp -r $CMSSW_BASE/src/CommonTools/Utils/test/ExpressionEvaluatorTests/ $CMSSW_BASE/src/ExpressionEvaluatorTests
> scram b ExpressionEvaluatorTests
> ls -l $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests/*/src
> $CMSSW_BASE/test/$SCRAM_ARCH/ExpressionEvaluatorUnitTest
> $CMSSW_BASE/test/$SCRAM_ARCH/testExpressionEvaluator
> rm -rf  $CMSSW_BASE/src/ExpressionEvaluatorTests
> |
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/7363#issuecomment-71833847>.
>

VinInn commented:
ok, so way need to be build at scram b time
ok
now assume I have the new "subsystem" in
CommonTools/Utils/test/ExpressionEvaluatorTests
can this be build and produce pch under
$CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests
(even $CMSSW_BASE/include/$SCRAM_ARCH/CommonTools/Utils/test/ExpressionEvaluatorTests ..)
how we do it for classes_def.xml?

smuzaffar commented:
currently pch  are only generated for Subsystem/Package/src/precompile.h 
files. I would suggest to put your precompiler.h in the 
CommonTools/Utils/src directory (along with the other headers needed by it).


On 1/28/2015 5:07 PM, Vincenzo Innocente wrote:
>
> ok, so way need to be build at scram b time
> ok
> now assume I have the new "subsystem" in
> CommonTools/Utils/test/ExpressionEvaluatorTests
> can this be build and produce pch under
> $CMSSW_BASE/include/$SCRAM_ARCH/ExpressionEvaluatorTests
> (even 
> $CMSSW_BASE/include/$SCRAM_ARCH/CommonTools/Utils/test/ExpressionEvaluatorTests 
> ..)
> how we do it for classes_def.xml?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/7363#issuecomment-71861174>.
>

cmsbuild commented:
+1
Tested at: a849aa813d8fbafde4c208e17c4c51dc4f02bb3e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2191/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7363/2191/summary.html

VinInn commented:
I would like to avoid possible undesired/circular dependency.
I think I can locate "precompile.h" in CandUtils or RecoUtils

VinInn committed:
Merged ExpressionEvaluator from repository VinInn

VinInn committed:
added very generic expression and index filter

VinInn commented on CommonTools/Utils/src/ExpressionEvaluator.cc:
```diff
@@ -0,0 +1,141 @@
+
+
+  source += "extern " + quote+'C'+quote+' ' + std::string(iname) + "* "+factory+"() {\n";
+  source += "static "+m_name+" local;\n";
```
add const...

VinInn commented on CommonTools/Utils/src/ExpressionEvaluator.cc:
```diff
@@ -0,0 +1,141 @@
+  }
+
+  m_expr = dlsym(dl,factory.c_str());
+
```
remove here

VinInn committed:
Unit test uses genericExpression, macro fixed for commas

VinInn committed:
stress test MT

VinInn committed:
move precompile.h in the release

VinInn committed:
use an existing dict for pat user data

VinInn committed:
Merged ExpressionEvaluator2 from repository VinInn

VinInn commented:
ooiops need rebase
ok let's close this one and reopen a new one
